server/utils/users/permissions.util.js

```diff
@@ -0,0 +1,43 @@
+import { verifyToken, Response } from '../index';
+
+const signInAuth = async (req, res, next) => {
+  const authHeader = req.headers.authorization;
+  if (typeof authHeader === 'undefined') {
+    return Response(res, 401, 'Unauthorized - Header Not Set');
+  }
+
+  const token = authHeader.split(' ')[1];
+  if (!token) {
+    return Response(res, 401, 'Access Denied. Please Log In.');
+  }
+
+  try {
+    const decoded = await verifyToken(token);
+    req.user = decoded;
+    next();
+  } catch (error) {
+    return Response(res, 500, 'Error in verification. Please try again');
```

Comment:
@TheDrizzyWay  Hey, error code 500  is for server errors, for verification should be either 401 or 403 which ever suites the purpose of the response

Reply:
@davidokonji Noted. will fix right away